### PR TITLE
risc-v/esp32: Fixed path of File-system mount point 

### PR DESCRIPTION
**Pull request by metarutaiga**
I tried to set custom path for my LittleFS but it's failed.

## Summary

the File-system Mount Point is miss in codes at my ESP32-C3.
So I fixed it and other ESP's risc-v codes. 

## Impact

Before
```
NuttShell (NSH) NuttX-12.6.0
nsh> ls
/:
 data/
 dev/
 proc/
 var/
```

After
```
NuttShell (NSH) NuttX-12.6.0
nsh> ls
/:
 dev/
 home/
 proc/
 var/
```

## Testing

* Set "/home" to mount point

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: boards/risc-v/esp32c3-legacy/common/src/esp32c3_board_spiflash.c
Comment (tmedicci) [inline boards/risc-v/esp32c3-legacy/common/src/esp32c3_board_spiflash.c:1]: It seems this file doesn't have any reference to `nuttx/config.h` (and that is probably the reason for https://github.com/apache/nuttx/actions/runs/11443502523/job/31836407638#step:7:720).

File: boards/risc-v/esp32c3/common/src/esp_board_spiflash.c
Comment (tmedicci) [inline boards/risc-v/esp32c3/common/src/esp_board_spiflash.c:1]: Same here (about the config).

File: boards/risc-v/esp32c6/common/src/esp_board_spiflash.c
Comment (tmedicci) [inline boards/risc-v/esp32c6/common/src/esp_board_spiflash.c:1]: Probably the same here.

File: boards/risc-v/esp32h2/common/src/esp_board_spiflash.c
Comment (tmedicci) [inline boards/risc-v/esp32h2/common/src/esp_board_spiflash.c:1]: Probably the same here.

File: boards/risc-v/esp32c3-legacy/common/src/esp32c3_board_spiflash.c
Comment (metarutaiga) [reply boards/risc-v/esp32c3-legacy/common/src/esp32c3_board_spiflash.c:1]: Yeah, I fixed it.

**Review by linguini1 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**